R/utils.R

```diff
@@ -35,8 +35,18 @@ wrapExpr <- function(func, ...) {
 #   expandExpr(quote(!!a + !!b), list(a = quote(two)), env)
 # })
 expandExpr <- function(expr, data, env) {
-  wrappedExpr <- wrapExpr(rlang::quo, expr)
-  rlang::quo_get_expr(eval(wrappedExpr, data, env))
+  do.call(bquote, list(expr = expr, where = env))
+}
+
+cleanExpr <- function(expr) {
+  walk_ast(expr, function(x) {
+    if (rlang::is_call(x, ".")) {
+      # TODO: stop if this call doesn't have a single, unnamed argument
+      cleanExpr(x[[2]])
+    } else {
+      cleanExpr(x)
+    }
+  })
 }
```

Comment:
Were you thinking we need to throw an error? I think I'd rather do something like this until we have a reason to disallow named/multiple arguments

```r
walk_ast(expr, function(x) {
    if (rlang::is_call(x, ".", n = 1) && rlang::is_call(x[[2]]) && length(x[[2]]) == 1 && identical(x[[2]], x[[2]][1])) {
      cleanExpr(x[[2]])
    } else {
      cleanExpr(x)
    }
  })
```

Reply:
I like the idea of leaving the node alone if it's not a single unnamed argument. But is your logic correct? It looks to me like it's testing if the child is of the form `foo()`, but there's no reason why we'd want to limit the contents to that; `.(input$xvar)` or `.(format(input$date))` should be allowed as well. It's things like `.()`, `.(expr = x)`, and `.(x, y)` that should be disallowed or ignored.

Reply:
Ahh, right, so it should be this?

```r
```r
walk_ast(expr, function(x) {
    if (rlang::is_call(x, ".", n = 1) && is.null(names(x))) {
      cleanExpr(x[[2]])
    } else {
      cleanExpr(x)
    }
  })
```
```

Reply:
Yeah that looks right. I didn't know about calling `names()` on a call object, that's nice.